api/v1beta1/azureclusteridentity_webhook.go

```diff
@@ -40,7 +43,18 @@ func (c *AzureClusterIdentity) ValidateCreate() error {
 
 // ValidateUpdate implements webhook.Validator so a webhook will be registered for the type.
 func (c *AzureClusterIdentity) ValidateUpdate(oldRaw runtime.Object) error {
-	return c.validateClusterIdentity()
+	var allErrs field.ErrorList
+	old := oldRaw.(*AzureClusterIdentity)
```

Comment:
Do we want this type assertion to possibly `panic`? That is, shouldn't we do:

```golang
old, ok := oldRaw.(*AzureClusterIdentity)
if !ok {
```

Reply:
I followed the existing way to maintain the consistency. I see that nowhere it is done like that. If you have strong opinions on that, I can refactor all of that as part of next PR.

Reply:
You're right that we do this consistently. The actual code paths must always pass a `runtime.Object` of the expected type, otherwise we'd be seeing `panic`s.

I think this approach isn't ideal, but I agree it's not in the scope of this PR. I'll think about it and create an issue.

Reply:
See #3683.